libraries/AP_Scripting/applets/switch_rates.lua

```diff
@@ -0,0 +1,176 @@
+  // @User: Standard
+--]]
+local PREV_RC_FUNC     = bind_add_param('RC_FUNC',       2, 300)
+
+-- params dictionary indexed by name
+local param_table = {
+   "ATC_ACCEL_P_MAX",
+   "ATC_ACCEL_R_MAX",
+   "ATC_ACCEL_Y_MAX",
+   "ATC_ANG_PIT_P",
+   "ATC_ANG_RLL_P",
+   "ATC_ANG_YAW_P",
+   "ATC_RAT_PIT_P",
+   "ATC_RAT_PIT_I",
+   "ATC_RAT_PIT_D",
+   "ATC_RAT_PIT_D_FF",
+   "ATC_RAT_PIT_FLTD",
+   "ATC_RAT_RLL_P",
+   "ATC_RAT_RLL_I",
+   "ATC_RAT_RLL_D",
+   "ATC_RAT_RLL_D_FF",
+   "ATC_RAT_RLL_FLTD",
+   "ATC_RAT_YAW_P",
+   "ATC_RAT_YAW_I",
+   "ATC_RAT_YAW_D",
+   "ATC_RAT_YAW_D_FF",
+   "ATC_RAT_YAW_FLTD",
+   "ATC_THR_G_BOOST",
+   "ACRO_RP_RATE_TC",
+   "ACRO_Y_RATE_TC",
+   "FSTRATE_ENABLE",
+   "FSTRATE_DIV",
+   "MOT_SPIN_MIN",
+   "MOT_SPIN_MAX",
+   "MOT_THST_EXPO",
+   "SERVO_DSHOT_RATE",
+}
+
+-- setup script specific parameters
+param:add_table(PARAM_BACKUP_TABLE_KEY, PARAM_BACKUP_TABLE_PREFIX, #param_table)
+
+local params = {}
+local prev_params = {}
+local param_count = 0
+
+if PREV_ENABLE:get() == 0 then
+   gcs:send_text(MAV_SEVERITY.NOTICE, string.format("Rate switch: disabled"))
+   return
+end
+
+local function add_param(pname)
+   local p = bind_param(pname)
+   if p then
+      params[pname] = p
+      param_count = param_count + 1
+      local px = bind_add_backup_param(pname, param_count, p:get())
+      if px then
+         prev_params[pname] = px
+      end
+   end
+end
+
+for _, pname in pairs(param_table) do
+   add_param(pname)
+end
+
+local function switch_parameters()
+   local count = 0
+   for pname, p1 in pairs(params) do
+      local p2 = prev_params[pname]
+      local v1 = p1:get()
+      local v2 = p2:get()
+      if v1 ~= v2 then
+         p1:set_and_save(v2)
+         p2:set_and_save(v1)
+         count = count + 1
+      end
+   end
+   return count
+end
+
+gcs:send_text(MAV_SEVERITY.INFO, string.format("Rate switch: stored %u parameters", param_count))
+
+local AuxSwitchPos = {LOW=0, MIDDLE=1, HIGH=2}
+local AuxSwitchPosNames = {"LOW", "MIDDLE", "HIGH"}
+
+-- start in LOW start
+local prev_pos = AuxSwitchPos.LOW
+
+-- main update function
+function update()
+   local sw_pos = rc:get_aux_cached(PREV_RC_FUNC:get())
+
+   if sw_pos ~= nil and sw_pos ~= prev_pos then
+      count = switch_parameters()
+      gcs:send_text(MAV_SEVERITY.INFO, string.format("Rate switch: %u parameters changed to %s", count, AuxSwitchPosNames[sw_pos + 1]))
+      prev_pos = sw_pos
+   end
+end
+
+-- wrapper around update(). This calls update() at 10Hz,
+-- and if update faults then an error is displayed, but the script is not
+-- stopped
+function protected_wrapper()
+  local success, err = pcall(update)
+  if not success then
+     gcs:send_text(MAV_SEVERITY.EMERGENCY, "Rate switch: internal Error: " .. err)
+     -- when we fault we run the update function again after 1s, slowing it
+     -- down a bit so we don't flood the console with errors
+     --return protected_wrapper, 1000
+     return
```

Comment:
If you just return here you might as well not bother with the `protected_wrapper` at all.